src/compiler/factory/nodeFactory.ts

```diff
@@ -7446,16 +7445,16 @@ function mergeEmitNode(sourceEmitNode: EmitNode, destEmitNode: EmitNode | undefi
 
     // `annotatedNodes` are not merged as they should only present on the parse tree node of a `SourceFile`.
 
-    // `leadingComments` are concatenated with any existing leading comments on the destination
+    // `leadingComments` overwrites the destination
     if (leadingComments) {
         // We use `.slice()` in case `destEmitNode.leadingComments` is pushed to later
-        destEmitNode.leadingComments = addRange(leadingComments.slice(), destEmitNode.leadingComments);
+        destEmitNode.leadingComments = leadingComments.slice();

```

Comment:
This code should be able to be removed now that main has the merging fixed.

Reply:
Done.